lib/ui/widgets/sleep_selector.dart

```diff
@@ -0,0 +1,206 @@
+                Navigator.of(context).pop(false);
+              },
+              child: Text(
+                texts.cancel_button_label,
+                style: TextStyle(color: theme.buttonColor),
+              ),
+            )
+          ],
+          content: Column(
+            mainAxisSize: MainAxisSize.min,
+            children: List<Widget>.generate(
+              options.length,
+              (int index) => dialogItemTile(
+                context,
+                theme,
+                texts,
+                audioBloc,
+                sleepPolice,
+                options[index],
+              ),
+            ),
+          ),
+        );
+      },
+    );
+  }
+
+  void changeSleepPolice(
+    BuildContext context,
+    L texts,
+    AudioBloc audioBloc,
+    SleepPolice newPolice,
+  ) {
+    audioBloc.changeSleepPolice(newPolice);
+    final scaffoldMessenger = ScaffoldMessenger.of(context);
+    if (scaffoldMessenger != null) {
+      scaffoldMessenger.showSnackBar(
+        SnackBar(
+          content: Text(
+            newPolice is SleepPoliceOff
+                ? texts.sleep_episode_function_toggled_off
+                : texts.sleep_episode_function_toggled_on,
+          ),
+        ),
+      );
+    }
+    Navigator.pop(context);
+  }
+
+  List<SleepOption> sleepOptions(L texts, SleepPolice sleepPolice) {
+    var options = <SleepOption>[];
+    if (!(sleepPolice is SleepPoliceOff)) {
+      options.add(SleepOption(
+        texts.sleep_episode_function_turn_off,
+        sleepPoliceOff(),
+      ));
+    }
+    options.addAll(
+      [
+        SleepOption(
+          texts.sleep_episode_function_15_minutes,
+          sleepPoliceMinutes(15),
+        ),
+        SleepOption(
+          texts.sleep_episode_function_30_minutes,
+          sleepPoliceMinutes(30),
+        ),
+        SleepOption(
+          texts.sleep_episode_function_45_minutes,
+          sleepPoliceMinutes(45),
+        ),
+        SleepOption(
+          texts.sleep_episode_function_60_minutes,
+          sleepPoliceMinutes(60),
+        ),
+      ],
+    );
+    return options;
+  }
+
+  Widget dialogItemTile(
+    BuildContext context,
+    ThemeData theme,
+    L texts,
+    AudioBloc audioBloc,
+    SleepPolice groupValue,
+    SleepOption option,
+  ) {
+    return InkWell(
+      highlightColor: theme.buttonColor.withAlpha(50),
+      onTap: () {
+        changeSleepPolice(context, texts, audioBloc, option.police);
+      },
+      child: Container(
+        padding: const EdgeInsets.fromLTRB(16.0, 8.0, 16.0, 8.0),
+        child: Row(
+          children: <Widget>[
+            Radio<SleepPolice>(
+              groupValue: groupValue,
+              value: option.police,
+              onChanged: (newValue) {
+                if (newValue != null) {
+                  changeSleepPolice(context, texts, audioBloc, newValue);
+                }
+              },
+              fillColor: dialogItemTileRadioFillColor(theme),
```

Comment:
Can't we inject that from the theme instead of creating an abstraction?

Reply:
@roeierez I'm not sure if it is possible, maybe you can help make it through theme data.

The first problem is the radio bullet should have a different color if it is selected or not, and these colors are not given by the same color slots through AnytimePlayer and Breez, specifically Breez dark.

Check this table, so you can see which colors are being used:

|-|Selected|Unselected|
|---|---|---|
|AP Light| buttonColor | unselectedWidgetColor |
|AP Dark| buttonColor | unselectedWidgetColor |
|Breez Light| buttonColor | unselectedWidgetColor |
|Breez Dark| buttonColor | buttonColor |

The second problem is the default theme for the radio is not correct, so to make it through ThemeData we need somehow add a `MaterialStateProperty<Color>` on theme data of anytime player and override it on the theme data of Breez, is it possible?

Change the unselectedWidgetColor of Breez dark has side-effects on others components so I guess it is not an option 🤔

Reply:
Not sure I understood everything but you don't have to override the top level themes. You can wrap locally with Theme widget which overiide any theme property you need.
https://api.flutter.dev/flutter/material/Theme-class.html
Does it help?

Reply:
🤔  If it is possible to override the theme locally for this specific widget, maybe it is helpful, let me study it a little bit and check if it is possible.

Reply:
@roeierez I did some things but nothing seems to work.

I did add the `radioTheme` on `ThemeData` and it works here at AnytimePlayer, the problem is on Breez, on Breez the colors on the radio are the ones defined one AnytimePlayer instead of getting from Breez ThemeDara.

What I did:

I did add the `radioTheme` for both `_buildLightTheme()` and `_buildDarktheme()` and both projects, AnytimePlayer and Breez

```dart
ThemeData _buildLightTheme() {
…
radioTheme: base.radioTheme.copyWith(
      fillColor: MaterialStateProperty.resolveWith(
…
      ),
    ),
…
```

To apply it I add on the Radio:

```
fillColor: Theme.of(context).radioTheme.fillColor,
```

it works fine for light and dark on the Anytime, but when I run on Breez it gets the Anytime theme colors.

any idea how to solve it? To be honest, I don't understand why the ThemeData of that context is not the app's theme data, it should work.

|Notice the orange instead of the blue|
|---|
|<img src="https://user-images.githubusercontent.com/1225438/133988664-27569d87-9cc8-4a66-bbcd-8b67d3fd4f02.png" alt="drawing" width="250"/>|